test/integration/authorizeRoles.test.js

```diff
@@ -0,0 +1,210 @@
+const chai = require("chai");
+const { expect } = chai;
+
+const { authorizeRoles } = require("../../middlewares/authorizeRoles");
+const authenticate = require("../../middlewares/authenticate");
+const authService = require("../../services/authService");
+const addUser = require("../utils/addUser");
+const cleanDb = require("../utils/cleanDb");
+const config = require("config");
+const cookieName = config.get("userToken.cookieName");
+const userData = require("../fixtures/user/user")();
+
+const defaultUser = userData[0]; // user with no `roles` key
+const appOwner = userData[3];
+const superUser = userData[4];
+
+// Route setup with various permissions for testing
+const express = require("express");
+const router = express.Router();
+const AppMiddlewares = require("../../middlewares");
+
+const pongHandler = (_, res) => {
+  return res.json({ message: "pong" });
+};
+
+router.get("/for-app-owner", authenticate, authorizeRoles(["app_owner"]), pongHandler);
+router.get("/for-super-user", authenticate, authorizeRoles(["super_user"]), pongHandler);
+router.get("/for-super-user-and-app-owner", authenticate, authorizeRoles(["super_user", "app_owner"]), pongHandler);
+router.get("/for-invalid", authenticate, authorizeRoles(["invalid"]), pongHandler);
+
+const app = express();
+AppMiddlewares(app);
+app.use("/", router);
+
+describe("authorizeRoles", function () {
+  let defaultJwt, appOwnerJwt, superUserJwt;
+
+  before(async function () {
+    const defaultUserId = await addUser(defaultUser);
+    const appOwnerId = await addUser(appOwner);
+    const superUserId = await addUser(superUser);
+    defaultJwt = authService.generateAuthToken({ userId: defaultUserId });
+    appOwnerJwt = authService.generateAuthToken({ userId: appOwnerId });
+    superUserJwt = authService.generateAuthToken({ userId: superUserId });
+  });
+
+  after(async function () {
+    await cleanDb();
+  });
+
+  describe("GET /for-app-owner", function () {
+    it("should authorize app owner for route with app_owner required role", function (done) {
+      chai
+        .request(app)
+        .get("/for-app-owner")
+        .set("cookie", `${cookieName}=${appOwnerJwt}`)
+        .end((err, res) => {
+          if (err) {
+            return done(err);
+          }
+          expect(res).to.have.status(200);
+          return done();
+        });
+    });
+
+    it("should not allow default user for route with app_owner required role", function (done) {
+      chai
+        .request(app)
+        .get("/for-app-owner")
+        .set("cookie", `${cookieName}=${defaultJwt}`)
+        .end((err, res) => {
+          if (err) {
+            return done(err);
+          }
+          expect(res).to.have.status(401);
+          return done();
+        });
+    });
+
+    it("should not authorize super user for route with app_owner required role", function (done) {
```

Comment:
Shouldn't super user be having access for app_owner routes too?

Reply:
No. This is specifically requested by @ankushdharkar to remove any form of role hierarchy.